prelude/slang-cuda-prelude.h

```diff
@@ -538,24 +564,63 @@ struct WaveOpMin
     __inline__ __device__ static T doOp(T a, T b) { return a < b ? a : b; }
 };
 
+template <typename T>
+struct ElementTypeTrait;
+
+// Scalar
+template <> struct ElementTypeTrait<int> { typedef int Type; };
+template <> struct ElementTypeTrait<uint> { typedef uint Type; };
+template <> struct ElementTypeTrait<float> { typedef float Type; };
+template <> struct ElementTypeTrait<double> { typedef double Type; };
+template <> struct ElementTypeTrait<uint64_t> { typedef uint64_t Type; };
+template <> struct ElementTypeTrait<int64_t> { typedef int64_t Type; };
+
+// Vector
+template <> struct ElementTypeTrait<int1> { typedef int Type; };
```

Comment:
I was going to ask why the vector case can't be handled via a partial specialization, but then I remembered that the vector types are built-ins for CUDA, while we have to implement matrices in user space.